native/swift/Sources/wordpress-api/WordPressLoginClient.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+import WordPressAPIInternal
+
+#if os(Linux)
+import FoundationNetworking
+#endif
+
+public final class WordPressLoginClient {
+
+    public protocol AuthenticatorProtocol {
+        func authenticate(url: URL, callbackURL: URL) async throws -> URL
+    }
+
+    private let requestExecutor: SafeRequestExecutor
+    private let client: UniffiWpLoginClient
+
+    public convenience init(
+        urlSession: URLSession,
+        middleware: MiddlewarePipeline = .default
+    ) {
+        self.init(requestExecutor: WpRequestExecutor(urlSession: urlSession), middleware: middleware)
+    }
+
+    init(
+        requestExecutor: SafeRequestExecutor,
+        middleware: MiddlewarePipeline = .default
+    ) {
+        self.requestExecutor = requestExecutor
+        self.client = UniffiWpLoginClient(requestExecutor: requestExecutor, middlewarePipeline: middleware)
+    }
+
+    func loginAttempt(
+        forSite proposedSiteUrl: String
+    ) async throws(AutoDiscoveryAttemptFailure) -> AutoDiscoveryAttemptResult {
+
+        let discoveryResult = await client.apiDiscovery(siteUrl: proposedSiteUrl)
```

Comment:
Can we change this function to `fn api_discovery(&self, site_url: String) -> Result<Arc<AutoDiscoveryAttemptSuccess>, AutoDiscoveryAttemptFailure>` and do the if-else checks below within Rust?

Reply:
The function currently returns `AutoDiscoveryResult/AutoDiscoveryUniffiResult` which is the result of combination of multiple attempts, so this suggestion wouldn't work. Happy to elaborate on it if you need more details.

Reply:
Yes. I'd love more details.

I haven't looked the how Kotlin uses the `api_discovery` function. But the Swift function here turns the `AutoDiscoveryUniffiResult` into either a success result or a failure, which essentially is `Result<Arc<AutoDiscoveryAttemptSuccess>, AutoDiscoveryAttemptFailure>`.